File: free-react-tailwind-admin-dashboard/src/types/salary.ts
// Backend response format (theo entities/salary.py)
export interface SalaryBackendResponse {
  employee_id: number;
  salary: number;        // Lương cơ bản
  allowance?: number;    // Phụ cấp  
  reward?: number;       // Thưởng
  detail?: string;       // Chi tiết/Ghi chú
  create_date: string;   // Ngày tạo
  update_date?: string;  // Ngày cập nhật
  
  // Thông tin nhân viên (nếu API trả về)
  employee_name?: string;
  employee_email?: string;
  employee_position?: string;
}

export interface SalaryCreate {
  employee_id: number;   // Backend format
  salary: number;        // Lương cơ bản
  allowance?: number;    // Phụ cấp
  reward?: number;       // Thưởng
  detail?: string;       // Chi tiết
}

export interface SalaryUpdate {
  salary?: number;       // Backend format
  allowance?: number;
  reward?: number;
  detail?: string;
}

File: backend/schemas/salary.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class SalaryBase(BaseModel):
    salary: int
    allowance: Optional[int] = None
    reward: Optional[int] = None
    detail: Optional[str] = None

class SalaryCreate(SalaryBase):
    employee_id: int

class SalaryUpdate(SalaryBase):
    salary: Optional[int] = None

class SalaryResponse(SalaryBase):
    employee_id: int
    create_date: datetime
    update_date: Optional[datetime] = None
    
    # Employee information
    employee_name: Optional[str] = None
    employee_email: Optional[str] = None
    employee_position: Optional[str] = None

    class Config:
        from_attributes = True

class SalaryListResponse(BaseModel):
    salaries: list[SalaryResponse]
    total: int
    page: int
    per_page: int
